client.go

```diff
@@ -49,10 +50,53 @@ func (resp *Response) IsUnauthorized() bool {
 	return resp.StatusCode() == http.StatusUnauthorized
 }
 
-func (client *Client) NewRequest(method, path string) *Request {
+type Opts struct {
+	Name   string
+	Digest string
+}
+
+type reqOpts struct {
+	Name   string
+	Digest string
+	UUID   string
+}
+
+type reqOpt func(o *reqOpts)
+
+func WithName(name string) reqOpt {
+	return func(o *reqOpts) {
```

Comment:
`...func(c *reqConfig) ` and `c.Name = ...` - this can be done for all the `With-` methods